packages/atomic/src/components/search/search-box-suggestions/suggestions-common.ts

```diff
@@ -3,41 +3,136 @@ import {VNode} from '@stencil/core';
 import {buildCustomEvent} from '../../../utils/event-utils';
 import {closest} from '../../../utils/utils';
 import {Bindings} from '../atomic-search-interface/atomic-search-interface';
+
+/**
+ * Element which will be rendered in the list of suggestions.
+ */
 export interface SearchBoxSuggestionElement {
+  /**
+   * Stable identity which enables Stencil to reuse DOM elements for better performance.
+   * The best way to pick a key is to use a string that uniquely identifies that list item among its siblings (often your data will already have IDs).
+   */
   key: string;
+  /**
+   * Rendered content of the element.
+   */
   content: Element | VNode;
+  /**
+   * Hook called when the selection is selected.
+   * @param e DOM event.
+   */
   onSelect?(e: Event): void;
-
+  /**
+   * The query associated with the suggestion, which will become the suggested query that would be sent if the search is executed.
+   */
   query?: string;
+  /**
+   * For improved accessibility, provide this property with additional information.
+   * https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label
+   */
   ariaLabel?: string;
+  /**
+   * Adds a specific shadow part attribute that can be selected with the CSS ::part pseudo-element.
+   * https://developer.mozilla.org/en-US/docs/Web/CSS/::part
+   */
   part?: string;
+  /**
+   * Hide the suggestion if it's the last in the list.
+   */
   hideIfLast?: boolean;
 }
+
+/**
+ * List of suggestions that will be displayed along other lists (e.g recent queries) when the search box's input is selected.
+ */
 export interface SearchBoxSuggestions {
+  /**
+   * The position of the list of the suggestions relative to the others.
+   */
   position: number;
+  /**
+   * Whether the suggestions should be listed in the right or left panel.
+   */
   panel?: 'left' | 'right';
+  /**
+   * Method that returns the list of elements which will be rendered in the list of suggestions.
+   */
   renderItems(): SearchBoxSuggestionElement[];
+  /**
+   * Hook called when the user changes the search box's input value.
```

Comment:
Similarly to the onSelect comment above, I think we should give an example of what to do here.

Reply:
The example code I will be using in the usage article utilizes this function. Do you think that should suffice or do we need an example here too?

Reply:
```suggestion
   * Hook called when the user changes the search box's input value. This can lead to all the query suggestions being updated.
```

Reply:
Yes I think the usage article should be linked in the reference doc, because it won't be enough to get a clear picture of what to do!